### PR TITLE
Define the GetDevicesUsage RPC

### PR DESCRIPTION
**Pull request by codingllama**
Add the means to fetch device usage stats from DeviceTrustService.

We may decide on a more centralized usage-acquiring mechanism in the future, but this should be alright for now.

https://github.com/gravitational/cloud/issues/5148

### CODE REVIEW

**Comment by codingllama** [comment]
FYI @sshahcodes, let me know if this looks alright.

**Review by flyinghermit — COMMENTED** [review]
Looks good.

File: api/proto/teleport/devicetrust/v1/devicetrust_service.proto [hunk 1]
```diff
@@ -142,6 +143,12 @@ service DeviceTrustService {
   // Authorized either by a valid MDM service certificate or the appropriate
   // "device" permissions (create/update/delete).
   rpc SyncInventory(stream SyncInventoryRequest) returns (stream SyncInventoryResponse);
+
+  // GetDevicesUsage retrieves devices limits and usage numbers for the
```
[the remark this answers is not in the document]
Comment (codingllama) [reply api/proto/teleport/devicetrust/v1/devicetrust_service.proto:147]: Done.

File: api/proto/teleport/devicetrust/v1/usage.proto [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.devicetrust.v1;
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1;devicetrustv1";
+
+// AccountUsageType is the type of the underlying account, either limitless or
```
[the remark this answers is not in the document]
Comment (codingllama) [reply api/proto/teleport/devicetrust/v1/usage.proto:21]: It's hard to say. Avoiding a boolean here is more of an API best practice than it is an immediate necessity. I'd usually recommend avoiding booleans unless we are 100% sure that whatever we are describing only ever has 2 states (and this includes not having an "undefined" state).

The weirdest part of this is having this enum sitting in devicetrust - that decision is short term, but I'm happy to rectify in the future.

https://protobuf.dev/programming-guides/api/#never-use-booleans-for-two-states

**Review by r0mant — APPROVED** [review]

**Review by flyinghermit — APPROVED** [review]

**Pull request merged**